### PR TITLE
ESLint configuration to help code with more style

### PR DESCRIPTION
**Pull request by michelgotta**
Overall JavaScript style was very mixed. ESLint was mostly bypassed with global and local ignores. Removed all those "bad" ignores.

To help write more "style-consistent" JS I fixed the ```npm run eslint``` command, the overall eslintconfig is a bit more forgivable and the code is more easy to read.

```npm run eslint``` runs without errors now matching all js files in ```assets/v2/js``` (Minified, already processed files, legacy and yge excluded).

### CODE REVIEW

**Comment by thelostone-mc** [comment]
Ah i knew this would result in lot of reviewing :P 
Will get to this over the weekend ^_^ but eslint config looks good

File: .eslintrc [hunk 1]
```diff
@@ -132,28 +124,15 @@
...
-      "error",
-      "always", {
-        "objectsInObjects": false
-      }
-    ],
-    "object-property-newline": "error",
     "one-var": [
       "error", {
-        "var": "always",
+        "var": "never",
```
Comment (thelostone-mc) [inline .eslintrc]: 👍

File: app/assets/v2/js/abi.js [hunk 2]
```diff
@@ -1,60 +1,49 @@
...
-    }
-}
+var bounty_address = function() {
+  if (document.web3network === null) {
+    // default to mainnet if web3network isn't found in time
+    document.web3network = 'mainnet';
+  }
+  switch (document.web3network) {
+    case 'mainnet':
+      return '0x2af47a65da8cd66729b4209c22017d6a5c2d2400';
```
Comment (thelostone-mc) [inline app/assets/v2/js/abi.js]: 👍 on removing the break

**Comment by michelgotta** [comment]
@thelostone-mc It's mostly style changes done with ```eslint --fix``` and some very slight refactorings. It looks like a lot of stuff but IMO it's fast to scan through the changes.

I think it's a good first step to a more consistent coding style.

**Comment by michelgotta** [comment]
Thanks for the feedback and the tip!

**Comment by mbeacom** [comment]
@michelgotta Hate to ask this, but would you mind rebasing this with `master` ? Thanks!

**Comment by michelgotta** [comment]
Let's for now try to merge the .eslintrc and package.json. I overwrote the conflicted files (someone is still working on) with the current master to make things easier. After the merge it will be much easier to add the unconsidered files.

**Comment by michelgotta** [comment]
@thelostone-mc @mbeacom I know there are a lot of changes coming from master, but is it realistic that these changes get into master? The "coding-style-mess" keeps me from working on this project. ;)

**Comment by thelostone-mc** [comment]
Haha if it can wait until #523 i'd be super happy :P 
But @michelgotta has been keeping this open for a while, so I can break my head with the conflicts if this needs to go first :)

**Comment by michelgotta** [comment]
@thelostone-mc I've seen all these changes in #523 and I think it's not very productive to spend a lot of time on my changes if the will become irrelevant after the #523 merge. I try to keep my changes rebased with the master.

**Comment by mbeacom** [comment]
@michelgotta Would you mind resolving the conflicts here?
Once the conflicts are resolved, I'll get this merged.

Thanks again for the contribution!

**Comment by michelgotta** [comment]
@mbeacom Merged with current master

File: app/assets/legacy/v2/js/amounts.js [hunk 3]
```diff
@@ -1,46 +1,49 @@
...
+    // store conv rate for later in cache
+    if (typeof document.conversion_rates == 'undefined') {
+      document.conversion_rates = {};
     }
+    document.conversion_rates[denomination] = conv_rate;
+    return callback(estimate(amount, conv_rate));
+  }).fail(function() {
+    return callback('Approx: Unknown amount');
+  });
+
```
Comment (thelostone-mc) [inline app/assets/legacy/v2/js/amounts.js]: @michelgotta Thoughts on removing the extra lines?

**Comment by michelgotta** [comment]
@thelostone-mc I guess this was unintentional. I struggled with a rebase/merge with sourcetree. There should not be any multi line deletions of code that makes sense.

**Comment by mbeacom** [comment]
@michelgotta Would you mind adding a snippet to the `./docs/CONTRIBUTING.md` outlining what's required to ensure a contributor's commits are compliant with these proposed rules?
Following merger of this PR, we can probably add this check to `pre-commit` and CI as well.

File: app/assets/legacy/v2/js/amounts.js (hunk 3 above)
Comment (michelgotta) [reply app/assets/legacy/v2/js/amounts.js]: My other comment was maybe a bit confusing. Viewed the diff on my phone and it looked odd. These lines should be deleted. Most of the changes were made with the auto-fix of eslint.

**Comment by mbeacom** [comment]
@michelgotta I'm good with changing the indentation for non-py files to 2 spaces.

**Comment by michelgotta** [comment]
@mbeacom For python and ruby code, I think 2 spaces are fine. With all the `{…}` I prefer 4 spaces (or better 1 tab) for indentation. It improves the legibility of the code and is a good standard.

**Comment by mbeacom** [comment]
@michelgotta Python is typically 4-space indentation with many references to this standard in PEP-8.  I'm fine with anything for js/html, but I know there have been discussions around using 2-space indentation on these files.

**Comment by michelgotta** [comment]
@mbeacom I looked around through popular repositories and I think 2-spaces is nowadays the most common for JS files. Changed the infos in the `CONTRIBUTING.md` and .eslintrc.

**Comment by michelgotta** [comment]
@thelostone-mc @mbeacom This branch is up to date with master. All necessary for JS style added.

**Comment by mbeacom** [comment]
@michelgotta 
screenshot 2018-03-13 10 29 06

Bounty creation doesn't appear to be working 🤔 The GET request to: `https://stage.gitcoin.co/api/v0.1/bounties/?github_url=https://github.com/mbeacom/test-repo/issues/22` returns `[]` as it should in the above case.  Current state of this branch is live at https://stage.gitcoin.co

**Comment by michelgotta** [comment]
@mbeacom Thanks for checking this. There are so major flaws with `!= null` and `== null` comparison. I have to recheck these points and revert the changes around the comparisons and the sanitize code in `shared.js`

**Comment by michelgotta** [comment]
@mbeacom I checked all changed if `someVar != null` and `someVar == null` parts in the js code and changed them back to `master`. Comparisons against `null`, `false` or `undefined` are always a struggle in js and hard to handle. I think it will work for now but there should be a type-safe checking at these points. Thanks again for testing and staging this branch!

**Comment by mbeacom** [comment]
@michelgotta Thanks for the contribution!

**Comment by michelgotta** [comment]
@mbeacom Thanks for the support and testing!

**Pull request merged**